cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkProgramRunner.java

```diff
@@ -204,10 +208,15 @@ public ProgramController run(Program program, ProgramOptions options) {
       }
 
       boolean isLocal = SparkRuntimeContextConfig.isLocal(options);
-      SparkSubmitter submitter = isLocal
-        ? new LocalSparkSubmitter()
-        : new DistributedSparkSubmitter(hConf, locationFactory, host, runtimeContext,
-                                        options.getArguments().getOption(Constants.AppFabric.APP_SCHEDULER_QUEUE));
+      SparkSubmitter submitter;
+      if (sparkConfig != null) {
+        submitter = new KubeSparkSubmitter(locationFactory, host, runtimeContext, sparkConfig);
```

Comment:
Kube is too specific over here. This is non-null whenever the MasterEnvironment is non-null, which just happens to only ever be kubernetes now, but may change in the futuree.

May be better to name this MasterEnvironmentSparkSubmitter or something like that.

Reply:
sounds good.